include/dlaf/blas_tile.tpp

```diff
@@ -40,6 +40,55 @@ void gemm(blas::Op op_a, blas::Op op_b, T alpha, const Tile<const T, device>& a,
              c.ptr(), c.ld());
 }
 
+template <class T, Device device>
+void hemm(blas::Side side, blas::Uplo uplo, T alpha, const Tile<const T, device>& a,
+          const Tile<const T, device>& b, T beta, const Tile<T, device>& c) {
+  SizeType m;
+  SizeType n;
+  m = c.size().rows();
+  n = c.size().cols();
+
+  if (side == blas::Side::Left) {
```

Comment:
I don't like switches: 1) awkward syntax; 2) if you forget a `break;` all hell breaks loose. Why would you prefer them?

Reply:
I completely agree with your points. What I found safer is that the compiler can warn you about a missing case for the enum.

The alternative I see is `if + [else if] + else`, with the "mandatory" `else` as a catch all. Maybe with `enum class` is even safe to just address all `if/else if cases` (but you may still missing some).

In this case, it is used `if (left) + if (right)` for `enum class Side   : char { Left     = 'L', Right    = 'R' };`, and I would at least change it to `if + else if`.

But well, I'm open to changing my mind 😁 I may be over-concerned about this, and the their awkwardness may easily overcome the added utility. Even more now that we use `enum class` instead of basic `enum`.

Looking forward to read your thoughts about this.

Reply:
> What I found safer is that the compiler can warn you about a missing case for the enum.

Yes, that's a good point. I find it less problematic in our case because we have few values/cases per enum (2-3), so I'm less concerned of missing a case.

Whatever is decided, I'm onboard. IMO we shouldn't spend too much time deciding between switch / if-else.